### PR TITLE
feat (GT-130): static modbus interface generation

### PR DESCRIPTION
**Pull request by gordon-epc**
Static modbus interface generation for grid-tied.

### CODE REVIEW

**Review by altendky — CHANGES_REQUESTED** [review]

File: src/epcpm/parameterstointerface.py [hunk 1]
```diff
@@ -32,6 +33,18 @@
...
 
+staticmodbus_types = {
+    "uint16": "sunsU16",
+    "enum16": "sunsU16",
+    "int16": "sunsS16",
+    "uint32": "sunsU32",
+    "int32": "sunsS32",
+    "string": "PackedString",
+    "bitfield16": "sunsU16",
+    "bitfield32": "sunsU32",
```
[the remark this answers is not in the document]
Comment (gordon-epc) [reply src/epcpm/parameterstointerface.py:45]: Added comment in c3c1694e82910cd9ee229743dddd85e94b23421c

File: src/epcpm/parameterstointerface.py [hunk 2]
```diff
@@ -319,25 +374,67 @@ def interface_variable_name(self):
...
+        # maybe_model = self.wrapped.tree_parent
+        #
+        # while not isinstance(maybe_model, epcpm.sunspecmodel.Model):
+        #     maybe_model = maybe_model.tree_parent
+        #
+        # model = maybe_model
+        # model_variable = f"sunspecInterface.model{model.id}"
+        #
+        # return f"&{model_variable}.{parameter.abbreviation}"
+        # # TODO: This needs to change to return staticmodbusInterface variable name, whenever that exists.
```
[the remark this answers is not in the document]
Comment (gordon-epc) [reply src/epcpm/parameterstointerface.py]: Updated comment + removed commented out code in c3c1694e82910cd9ee229743dddd85e94b23421c

File: src/epcpm/parameterstointerface.py [hunk 3]
```diff
@@ -1488,6 +1685,13 @@ def create_common_initializers(
             f".handSetter = {hand_coded_sunspec_setter_function},",
         ],
         f"}},",
+        f".staticmodbus = {{",
+        [
+            f".variable = {staticmodbus_variable},",
```
[the remark this answers is not in the document]
Comment (gordon-epc) [reply src/epcpm/parameterstointerface.py:1680]: No action taken. This will be handled when staticmodbus.common.variable is removed in the future.

**Review by altendky — APPROVED** [review]
As always, thanks for the work.  Except I'm not sure I actually say that here.  So, thank you.

**Pull request merged**